packages/components/src/modal/Modal.tsx

```diff
@@ -139,7 +151,11 @@ function Modal({
               >
                 <div
                   className="modal-content"
-                  onClick={e => e.stopPropagation()}
+                  onMouseDown={e => {
+                    e.stopPropagation();
+                    setContentClicked(true);
```

Comment:
Instead of doing `setContentClicked`, can you just do `setBackgroundClicked` when mousedown in the background? Then you don't need to listen to click, down, or up in the content